modules/channel-smooch/src/backend/client.ts

```diff
@@ -0,0 +1,128 @@
+import * as sdk from 'botpress/sdk'
+import Smooch from 'smooch-core'
+
+import { Config } from '../config'
+
+import { Clients } from './typings'
+
+const name = 'smooch.sendMessage'
+
+export class SmoochClient {
+  private smooch: any
+  private path: string
+
+  constructor(
+    private bp: typeof sdk,
+    private botId: string,
+    private config: Config,
+    private router: any,
+    private route: string
+  ) {}
+
+  async initialize() {
+    this.smooch = new Smooch({
+      keyId: this.config.keyId,
+      secret: this.config.secret,
+      scope: 'app'
+    })
+
+    const url = (await this.router.getPublicPath()) + this.route
```

Comment:
I think you could just type /webhook here instead of passing the variable this.route, there's no plus value of declaring it in the index file

Reply:
The index file uses the value of route to setup for the post requests
```suggestion
router.post(route, async (req, res) => {
```
I can stop passing route to the client but that would mean declaring it twice. Maybe handling of post requests should be in the client.ts file?